puffin/src/lib.rs

```diff
@@ -430,6 +455,14 @@ impl ThreadProfiler {
     }
 }
 
+/// Incremental monolithic counter to identify scopes.
+static SCOPE_ID_TRACKER: std::sync::atomic::AtomicU32 = std::sync::atomic::AtomicU32::new(0);
+
+fn fetch_add_scope_id() -> ScopeId {
+    let new_id = SCOPE_ID_TRACKER.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
```

Comment:
I guess this could be relaxed?
But it might not be hot enough to matter?

Reply:
i am not entirely sure, i can experience a bit but thought it would need to be quite restrict we don't want to have any duplicates.

Reply:
This can for sure be `Relaxed` - it has no dependency on other data. It can have a pretty big performance impact.